### PR TITLE
p2p: set RecvMessageCapacity to maxMsgSize in all reactors

### LINKED ISSUE #1503: Ensure each channel has capacity defined
```
func (bcR *BlockchainReactor) GetChannels() []*p2p.ChannelDescriptor {
	return []*p2p.ChannelDescriptor{
		{
			ID:                  BlockchainChannel,
			Priority:            10,
			SendQueueCapacity:   1000,
			RecvBufferCapacity:  50 * 4096,
			RecvMessageCapacity: maxMsgSize,
		},
	}
}
```

vs

```
func (memR *MempoolReactor) GetChannels() []*p2p.ChannelDescriptor {
	return []*p2p.ChannelDescriptor{
		{
			ID:       MempoolChannel,
			Priority: 5,
		},
	}
}
```

### PR DESCRIPTION
**Pull request by melekes**
to prevent malicious nodes from sending us large messages (~21MB, which
is the default `RecvMessageCapacity`)

This allows us to remove unnecessary `maxMsgSize` check in `decodeMsg`. Since each channel has a msg capacity set to `maxMsgSize`, there's no need to check it again in `decodeMsg`.

Closes #1503

### CODE REVIEW

File: mempool/reactor.go [hunk 1]
```diff
@@ -271,10 +273,6 @@ func RegisterMessages(cdc *amino.Codec) {
 }
 
 func (memR *Reactor) decodeMsg(bz []byte) (msg Message, err error) {
-	maxMsgSize := calcMaxMsgSize(memR.config.MaxTxBytes)
-	if l := len(bz); l > maxMsgSize {
-		return msg, ErrTxTooLarge{maxMsgSize, l}
```
Comment (erikgrinaker) [inline mempool/reactor.go:276]: Is this API breaking? Are there clients that may depend on `ErrTxTooLarge` being returned, in some form?

Comment (melekes) [reply mempool/reactor.go:276]:
```go
		memR.Logger.Error("Error decoding message", "src", src, "chId", chID, "msg", msg, "err", err, "bytes", msgBytes)
		memR.Switch.StopPeerForError(src, err)
```

no, we log this error and stop peer

Comment (melekes) [reply mempool/reactor.go:276]: this is a private function, no external clients depend on it

Comment (erikgrinaker) [reply mempool/reactor.go:276]: Yeah, I see that we have an additional check for this in the mempool.

**Pull request merged**